ocs_ci/utility/perf_dash/dashboard_api.py

```diff
@@ -0,0 +1,510 @@
+            return True
+        except Exception as err:
+            log.error(f"Can not execute [{query}]\n{err}")
+            return False
+
+    def get_id_by_name(self, table=None, name=None):
+        """
+        Query the ID of specific 'name' value from selected table
+
+        Args:
+            table (str): The table from which data will be query
+            name (str): The value for the 'name' field in the table to query
+
+        Returns:
+             int : the value of the ID field in the table, otherwise None
+
+        """
+        query = f"SELECT id FROM {table} WHERE name = '{name}' ;"
+        if self._run_query(query=query):
+            for result_id in self.cursor:
+                # The query return a tuple and we need only the firs element
+                return result_id[0]
+
+        return None
+
+    def get_name_by_id(self, table=None, recid=None):
+        """
+        Query the name of specific 'id' value from selected table
+
+        Args:
+            table (str): The table from which data will be query
+            recid (int): The value for the 'id' field in the table to query
+
+        Returns:
+             str : the value of the Name field in the table, otherwise None
+
+        """
+        query = f"SELECT name FROM {table} WHERE id = {recid} ;"
+        if self._run_query(query=query):
+            for result_name in self.cursor:
+                # The query return a tuple and we need only the first element
+                return result_name[0]
+
+        return None
+
+    def insert_single_value(self, table=None, value=None):
+        """
+        Insert a value to 'table' and return it's ID
+
+        Args:
+            table (str): The table to insert data into
+            value (str): The value to insert
+
+        Returns:
+             int : the ID of the value in the table, otherwise None
+
+        """
+        # prevent of duplicate records for the same build
+        record_id = self.get_id_by_name(table=table, name=value)
+        if record_id is not None:
+            return record_id
+
+        query = f"INSERT INTO {table} (name) VALUES ('{value}') ;"
+        if self._run_query(query=query):
+            try:
+                rec_id = self.cursor.lastrowid
+                # Make sure data is committed to the database
+                self.cnx.commit()
+                return rec_id
+            except Exception as err:
+                log.error(f"Can not insert {value} into {table} - [{err}]")
+                return None
+
+    def get_version_id(self, version=None):
+        """
+        Query of the ID of version number in the DB
+
+        Args:
+            version (str): The version number (e.g. 4.9.0)
+
+        Returns:
+             int : the version ID in the DB
+
+        """
+        return self.get_id_by_name("versions", version)
+
+    def get_test_id(self, test=None):
+        """
+        Query of the ID of test name in the DB
```

Comment:
It would be nice to add the table metadata(fields) to the class string so that its easier for others to follow the structure for the queries.

Reply:
Thanks for that @vasukulkarni , i just add a full database structure description in the beginning of the file